cli_tools/gce_vm_image_export/main_test.go

```diff
@@ -0,0 +1,157 @@
+	}
+}
+
+func TestFlagsSouceImageNotProvided(t *testing.T) {
+	defer testutils.BackupOsArgs()()
+	cliArgs := getAllCliArgs()
+	defer testutils.ClearStringFlag(cliArgs, sourceImageFlagKey, &clientID)()
+	buildOsArgsAndAssertErrorOnValidate(cliArgs, "Expected error for missing source_image flag", t)
+}
+
+func TestFlagsClientIdNotProvided(t *testing.T) {
+	defer testutils.BackupOsArgs()()
+	cliArgs := getAllCliArgs()
+	defer testutils.ClearStringFlag(cliArgs, clientIDFlagKey, &clientID)()
+	buildOsArgsAndAssertErrorOnValidate(cliArgs, "Expected error for missing client_id flag", t)
+}
+
+func TestFlagsDestinationUriNotProvided(t *testing.T) {
+	defer testutils.BackupOsArgs()()
+	cliArgs := getAllCliArgs()
+	defer testutils.ClearStringFlag(cliArgs, destinationUriFlagKey, &clientID)()
+	buildOsArgsAndAssertErrorOnValidate(cliArgs, "Expected error for missing destination_uri flag", t)
+}
+
+func buildOsArgsAndAssertErrorOnValidate(cliArgs map[string]interface{}, errorMsg string, t *testing.T) {
+	testutils.BuildOsArgs(cliArgs)
+	if err := validateAndParseFlags(); err == nil {
+		t.Error(errorMsg)
+	}
+}
+
+func TestBuildDaisyVarsWithoutFormatConversion(t *testing.T) {
+	defer testutils.SetStringP(&region, "aRegion")()
+	got := buildDaisyVars()
+
+	assert.Equal(t, "global/images/anImage", got["source_image"])
+	assert.Equal(t, "gs://bucket/exported_image", got["destination"])
+	assert.Equal(t, "global/networks/aNetwork", got["export_network"])
+	assert.Equal(t, "regions/aRegion/subnetworks/aSubnet", got["export_subnet"])
+	assert.Equal(t, 4, len(got))
+}
+
+func TestBuildDaisyVarsWithFormatConversion(t *testing.T) {
+	defer testutils.SetStringP(&region, "aRegion")()
+	defer testutils.SetStringP(&format, "vmdk")()
+
+	got := buildDaisyVars()
+
+	assert.Equal(t, "global/images/anImage", got["source_image"])
+	assert.Equal(t, "gs://bucket/exported_image", got["destination"])
+	assert.Equal(t, "vmdk", got["format"])
+	assert.Equal(t, "global/networks/aNetwork", got["export_network"])
+	assert.Equal(t, "regions/aRegion/subnetworks/aSubnet", got["export_subnet"])
+	assert.Equal(t, 5, len(got))
+}
+
+func TestPopulateMissingParametersDoesNotChangeProvidedScratchBucketAndUsesItsRegion(t *testing.T) {
+	defer testutils.SetStringP(&zone, "")()
+	defer testutils.SetStringP(&scratchBucketGcsPath, "gs://scratchbucket/scratchpath")()
+	defer testutils.SetStringP(&destinationUri, "gs://destbucket/destfile")()
+	defer testutils.SetStringP(&project, "a_project")()
+
+	err :=
+			paramutils.RunTestPopulateMissingParametersDoesNotChangeProvidedScratchBucketAndUsesItsRegion(
+				t, zone, region, scratchBucketGcsPath, destinationUri, project, "scratchbucket",  "europe-north1", "europe-north1-b")
+
+	assert.Nil(t, err)
+	assert.Equal(t, "europe-north1-b", *zone)
+	assert.Equal(t, "europe-north1", *region)
+	assert.Equal(t, "gs://scratchbucket/scratchpath", *scratchBucketGcsPath)
+}
+
+func TestPopulateMissingParametersCreatesScratchBucketIfNotProvided(t *testing.T) {
```

Comment:
this is dup by purpose. We want to test whether vars in importer/exporter main.go are actually populated (modified). As said in the PR description, if we only test the population function, it's not safe enough - maybe local vars in population function got populated but vars in main.go kept empty.

Reply:
PopulateMissingParameters has been moved to paramutils. That's why I want to keep the tests for main.go, otherwise there isn't any test cases verifying whether those flags are populated correctly for main program